go-controller/pkg/config/cni.go

```diff
@@ -71,29 +72,86 @@ func WriteCNIConfig() error {
 
 // ParseNetConf parses config in NAD spec
 func ParseNetConf(bytes []byte) (*ovncnitypes.NetConf, error) {
+	netconf, isRecoverable, err := parseNetConfList(bytes)
+	if err != nil {
+		if isRecoverable {
+			netconf, _, err = parseNetConfSingle(bytes)
+			if err != nil {
+				return nil, err
+			}
+		} else {
+			return nil, err
+		}
+	}
+
+	if netconf.Topology == "" {
+		// NAD of default network
+		netconf.Name = ovntypes.DefaultNetworkName
+	}
+
+	return netconf, nil
+}
+
+func parseNetConfSingle(bytes []byte) (*ovncnitypes.NetConf, bool, error) {
 	netconf := &ovncnitypes.NetConf{MTU: Default.MTU}
 	err := json.Unmarshal(bytes, &netconf)
 	if err != nil {
-		return nil, err
+		return nil, true, err
 	}
+
 	// skip non-OVN NAD
 	if netconf.Type != "ovn-k8s-cni-overlay" {
-		return nil, ErrorAttachDefNotOvnManaged
+		return nil, true, ErrorAttachDefNotOvnManaged
```

Comment:
how can this error be recoverable ? The user is requesting a different type of plugin.

Furthermore, I wonder if it indeed makes sense to have the concept of recoverable error for the `parseNetConfSingle` func. I don't think so.

The fact you ignore its value in the production code hints at that.

Reply:
This is the only method i found for unmarshal when the fields are omitEmpty
the unmarshel doesnt return an error, so this is what allows us to try the 2nd option

Reply:
`parseNetConfSingle` is invoked exactly **once** and its second return argument is **always** ignored.

Thus, why have it ? Do you have any plans to "recover from" the non-list configuration type ?... What to ? 

For clarity: the only function from which you **may** need to recover from is `parseNetConfList`. 

Am I missing something ?

Reply:
Right, i just made them interchangeable (in case we change the invocation order / to make it easier to read if they are symmetric)
But i will remove it, and keep it only for `parseNetConfList`